src/odemis/acq/millmng.py

```diff
@@ -0,0 +1,526 @@
+
+    except yaml.YAMLError as exc:
+        logging.error("Syntax error in milling settings yaml file: %s", exc)
+
+    return settings_dict
+
+
+# To handle the timeout error when the stage is not able to move to the desired position
+# It logs the message and raises the MoveError exception
+class MoveError(Exception):
+    pass
+
+
+class MillingRectangleTask(object):
+    """
+    This class represents a milling Task for milling rectangular regions on the sample.
+    """
+
+    def __init__(self, future: futures.Future, millings: list, sites: list, feature_post_status: str, acq_streams,
+                 ebeam, sed, stage, aligner, log_path=None):
+        """
+        Constructor
+        :param future: (ProgressiveFuture) the future that will be executing the task
+        :param millings: (list of MillingSettings) Settings corresponding to each milling, to be milled in order
+        :param sites: (list of Features) Each Feature to be milled
+        :param feature_post_status: (str) value to set on the Feature at the end of a complete milling series
+        :param acq_streams: type of acquisition streams to be used for milling
+        :param ebeam: model component for the scanner
+        :param sed: model component for the detector
+        :param stage: model component for the stage
+        :param aligner: model component for the aligner
+        :param log_path: (str) path to the log anchor region acquisition used in drift correction
+        """
+        self._stage = stage
+        self._scanner = ebeam
+        self._detector = sed
+        self._aligner = aligner
+
+        self._future = future
+        if future is not None:
+            self._future.running_subf = model.InstantaneousFuture()
+            self._future._task_lock = threading.Lock()
+
+        self._settings = millings
+        self._feature_status = feature_post_status  # site and feature means the same
+        self.sites = sites
+        self.streams = acq_streams
+
+        self._log_path = log_path
+        if log_path:
+            filename = os.path.basename(self._log_path)
+            if not filename:
+                raise ValueError("Filename is not found on log path.")
+            self._exporter = find_fittest_converter(filename)
+            self._fn_bs, self._fn_ext = dataio.splitext(filename)
+            self._log_dir = os.path.dirname(self._log_path)
+
+        # internal class requirements between functions
+        self._pass_duration = []  # one value per milling setting, duration(s) after which drift acquisition starts
+        self._iterations = []  # one value per milling setting, numbers of scans of .roi
+        stage_md = stage.getMetadata()
+        try:
+            self._ion_beam_angle = stage_md[model.MD_ION_BEAM_TO_SAMPLE_ANGLE]
+        except KeyError:
+            raise ValueError("Ion beam angle not defined in stage metadata")
+        self._move_speed = MOVE_SPEED_DEFAULT  # stage speed used from _tiledacq.py
```

Comment:
Contrarily to what is suggested by the comment, that's *not* how tiledacq.py does estimate the move speed. Please take the whole code from there. It starts by trying to get the speed reported by the stage. MOVE_SPEED_DEFAULT is only a fallback.